atc/exec/task_step.go

```diff
@@ -264,25 +264,20 @@ func (step *TaskStep) run(ctx context.Context, state RunState, delegate TaskDele
 		step.lockFactory,
 	)
 
-	if err != nil {
-		if err == context.Canceled || err == context.DeadlineExceeded {
-			step.registerOutputs(logger, repository, config, result.VolumeMounts, step.containerMetadata)
-		}
-		return false, err
-	}
-
-	delegate.Finished(logger, ExitStatus(result.ExitStatus))
-
```

Comment:
At first I was hesitant about caching stuff from an errored task since it could've left the cache in an invalid state. But then I realized we also register caches on a failed task (which could also leave the cache invalid). Guess it's just up to the user to deal with potentially corrupted caches

Reply:
Yeah, I think this is worth the trade-off since it's annoying if e.g. a task fetches a bunch of dependencies and then flakes out or something. We'll see if it's problematic. There's a `fly clear-task-cache` command now which should also help.